src/04-extensions.js

```diff
@@ -0,0 +1,212 @@
+  let indexToChopOffRemainingRooms;
+  let gottenPath = "";
+
+  //iterate through path until we are out of bounds
+  for (let i = 0; i < selectedPath.length; i++) {
+    //if next index is out of bounds return true cause we've hit the end of path successfully
+    if (selectedPath[i + 1] === undefined) break;
+    //declare flag variable which will decide if we proceed so long as next room is valid
+    let nextRoomIsValid = false;
+
+    for (let j = 0; j < remainingRooms.length; j++) {
+      if (remainingRooms[j].roomId === lastRoom) {
+        if (remainingRooms[j].connectsTo.includes(selectedPath[i + 1])) {
+          //set next room is valid to true
+          nextRoomIsValid = true;
+          //re-assign index to chop off to current index post-iteration
+          indexToChopOffRemainingRooms = j;
+          lastRoom = selectedPath[i + 1];
+          if (i === selectedPath.length - 2) {
+            gottenPath += `${remainingRooms[j].name} -> `;
+            gottenPath += `${remainingRooms[j + 1].name} -> `;
+            break;
+          } else {
+            gottenPath += `${remainingRooms[j].name} -> `;
+            break;
+          }
+        }
+      }
+    }
+    if (!nextRoomIsValid) return false;
+    remainingRooms.splice(indexToChopOffRemainingRooms, 1);
+  }
+  if (showPath) return gottenPath.slice(0, -3);
+  return true;
+}
+// console.log(validateMuseumPath(valid, true));
+
+/*
+ticketExtraDetails()
+Write a new function called ticketExtraDetails() that takes a string as its argument. If that string matches 
+one of the ticket "extras", provide information about what someone would be able to do based on purchasing 
+that ticket. The format is up to you.
+*/
+
+function ticketExtraDetails(extras) {
+  //edge case - if the extra is not movie education or terrace return error message
+  const validExtras = ["movie", "education", "terrace"];
+
+  if (!Array.isArray(extras) && !validExtras.includes(extras.toLowerCase())) {
+    return "must be a valid extra: 'movie', 'education', or 'terrace'";
+  } else {
+    extras = extras.toLowerCase();
+  }
+
+  //if argument is not an array and data type is string
+  if (!Array.isArray(extras) && typeof extras === "string") {
+    if (extras === "education") return "If you purchase this ticket extra you will gain access to 2 more rooms and see 2 more dinosaurs!";
+    if (extras === "terrace") return "If you purchase this ticket extra you will gain access to 1 more room!";
+    if (extras === "movie") return "If you purchase this ticket extra you will gain access to 3 more rooms and see 2 more dinosaurs and see 1 jurassic park movie!";
+  }
+
+  //purpose of creating object is to create reusable code for if we ever want to add extras to extras
+  const extrasObj = {
+    //ordered list represents access to x more rooms, see x more dinos, and see 1 jurassic park movie
+    education: [2, 2, 0],
+    terrace: [1, 0, 0],
+    movie: [3, 2, 1],
+  };
```

Comment:
Ideally this would be dynamically generated as opposed to hardcoded. Data can (and will!) change in real applications.